File: Difficulty: Easy/Check K-th Bit/check-kth-bit.java
class CheckBit {
    static boolean checkKthBit(int n, int k) {
        // code here
        String binary = "";
        while(n > 0){
            binary = (n % 2) + binary;
            n /= 2;
        }
        
        
        if(k >= binary.length()) return false;
        // System.out.println(binary);
        k = binary.length() - k - 1;
        
        
        
        // System.out.println(k);
        
            if(binary.charAt(k) == '1'){
                return true;
            }
        
        return false;
    }
}

// 100

File: Difficulty: Easy/Check K-th Bit/check-kth-bit.cpp
class Solution {
  public:
    bool checkKthBit(int n, int k) {
        //  code here
        // string s="";
        // while(n>0){
        //     s=to_string(n%2)+s;
        //     n/=2;
        // }
        // if(k>=s.size())return false;
        
        // k = s.size()-k-1;
        // if(s[k]=='1')return true;
        
        //      return false;
        n=n>>k;
        return (n&1);
    }
};


//main h k=length-k-1

// n=n<<k;
// return (n&1);